app/controllers/taxons_controller.rb

```diff
@@ -92,8 +92,7 @@ def destroy
       flash.now[:danger] = t("controllers.taxons.destroy_no_redirect")
       render :confirm_delete, locals: { page: Taxonomy::DeletePage.new(taxon) }
     else
-      base_path = Services.publishing_api.get_content(params[:taxon][:redirect_to])['base_path']
-      Services.publishing_api.unpublish(params[:id], type: "redirect", alternative_path: base_path)
+      UnpublishTaxonWorker.perform_async(params[:id], params[:taxon][:redirect_to])
```

Comment:
Why make this asynchronous? It makes it a lot harder to provide feedback to the user if this fails.

Reply:
Because it loops through a possibly large number of content items and tags them to the parent - this may involve a large number of calls to the publishing api. If I make this synchronous, the user may have to wait a long time.

Reply:
How about mirroring the manual approach, and using TagMigration stuff? This has progress reporting support, and works asynchronously.

Reply:
@cbaines I'd be interested to see how this works.

Reply:
fixed